pkg/auth/auth.go

```diff
@@ -120,12 +123,17 @@ type Config struct {
 }
 
 func newHTTPClient(issuerCA string, includeSystemRoots bool) (*http.Client, error) {
-	if issuerCA == "" {
-		return http.DefaultClient, nil
-	}
-	data, err := ioutil.ReadFile(issuerCA)
+	data, err := ioutil.ReadFile(defaultIssuerCA)
```

Comment:
This isn't exactly right.

* We want to use the service account ca.crt for OAuth metadata discovery (`.well-known/oauth-authorization-server`), but *not* use the proxy trusted CA bundle for metadata discovery
* We don't want to use service account ca.crt for OIDC, only OpenShift auth

This helper is called from different places, and we only want to include service account ca.crt is only included when necessary.

Reply:
Right, do we want to include the `trusted-ca-bundle` in all the cases ?